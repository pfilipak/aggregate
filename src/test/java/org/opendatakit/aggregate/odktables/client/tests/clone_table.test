# Set up
createUser user1
createUser user2

createSynchronizedTable user1 people
    name STRING false
    age INTEGER false
    weight DECIMAL false


# Test that user can clone a table he has read permission on

setTablePermissions user1 user2 people true false false
cloneSynchronizedTable user2 people


# Test that user can't clone a table more than once
cloneSynchronizedTable user2 people


# Test that user can't clone a table that doesn't exist
cloneSynchronizedTable user2 nonexistenttable


# Test that user can't clone a table that he doesn't have read permission on
