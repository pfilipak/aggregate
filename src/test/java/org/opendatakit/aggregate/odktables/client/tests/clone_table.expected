# Set up
createUser user1
createUser user2

createSynchronizedTable user1 people


# Test that user can clone a table he has read permission on
setTablePermissions user1 user2 people true false false
cloneSynchronizedTable user2 people


# Test that user can't clone a table more than once
cloneSynchronizedTable user2 people
Exception: org.opendatakit.aggregate.odktables.client.exception.TableAlreadyExistsException: Table with tableID 'people' already exists!


# Test that user can't clone a table that doesn't exist
cloneSynchronizedTable user2 nonexistenttable
Exception: org.opendatakit.aggregate.odktables.client.exception.TableDoesNotExistException: Table with ID nonexistenttable does not exist!


# Test that user can't clone a table that he doesn't have read permission on
listAllTables user2
tableName: people, userName: user1

removeTableSynchronization user2 people
setTablePermissions user1 user2 people false false false

listAllTables user2

cloneSynchronizedTable user2 people
Exception: org.opendatakit.aggregate.odktables.client.exception.TableDoesNotExistException: Table with ID people does not exist!


# Clean up
deleteSynchronizedTable user1 people
deleteUser user1
deleteUser user2